In [24]:
import pandas as pd
import pprint
pp = pprint.PrettyPrinter(indent =4)

I will use the CSV files created in part 2 of my project. The following files were created:
1. Data from 2014
2. Scraped data from 2021

In [27]:
Amazon_beauty_data= pd.read_csv('Final_Beauty_Data.csv')

In [28]:
Amazon_beauty_data.head()

Unnamed: 0        asin  helpful  Not helpful  \
0           0  B00KWE08Q0      0.9          0.9   
1           1  B004W75SLI      1.6          3.2   
2           2  B00JL2TURM      1.3          1.6   
3           3  B003WNXZZO      3.8          5.6   
4           4  B001B7GJ2E      1.7          2.2   

                                       review_concat  \
0  love brush I never tried mini kabuki brush one...   
1  great mask clear skin never tried type mask re...   
2  received product consideration tomoson like id...   
3  unfortunately burn skinnormal rose water doesn...   
4  amazing tool hat like little curve end side ma...   

                                      summary_concat  overall rating  \
0  love brush great brush perfect countouring gre...               5   
1  five star awesome improvement not clear enough...               5   
2  nice natural exfoliant freaking cool amazing w...               5   
3  unfortunately burn skin normal rose water not ...               1   
4  amazing attention mom help teen prevent pockma...               5   

                                                name             category  \
0  Mini Kabuki Makeup Brush Set – Beauty Junkees ...  Tools & Accessories   
1  Origins Clear Improvement Active Charcoal Mask...            Skin Care   
2  pureSOL Konjac Sponge - Activated Charcoal - F...        Personal Care   
3  Shea Terra Organics Beauty Water, Toner & Face...            Skin Care   
4       Body Toolz Double Loop Facial Skin Care Tool  Tools & Accessories   

                                         description  price  rating  \
0  Brand\n\n\nBeauty Junkees\n\n\n\n\nColor\n\n\n...  19.97     4.6   
1  Brand\n\n\nOrigins\n\n\n\n\nSkin Type\n\n\nAll...  14.25     4.5   
2  Color\n\n\nBamboo Charcoal\n\n\n\n\nBrand\n\n\...  10.00     4.2   
3  Ingredients\n\n\nPure Moroccan rose water, Est...  15.75     4.3   
4  Brand\n\n\nBody Toolz\n\n\n\n\nMaterial\n\n\nS...   7.75     4.1   

   No_of_Rating  
0          1366  
1          1082  
2          2466  
3           120  
4           673

In [29]:
Amazon_beauty_data.shape

(4242, 13)

In [30]:
print('These are the features from dataset')
print(Amazon_beauty_data.columns)

These are the features from dataset
Index(['Unnamed: 0', 'asin', 'helpful', 'Not helpful', 'review_concat',
       'summary_concat', 'overall rating', 'name', 'category', 'description',
       'price', 'rating', 'No_of_Rating'],
      dtype='object')


There are 11 features for each asin in the dataset, out of which 5 features, 'helpful', 'Not helpful', 'review_concat','summary_concat', 'overall rating' are from old dataset 

6 features, 'name', 'category', 'description','price', 'rating', 'No_of_Rating' are the scraped data

I will now look at the datatypes and null values of each column and see if something needs to be changed or dropped

In [39]:
Amazon_beauty_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4242 entries, 0 to 4241
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   asin            4242 non-null   object 
 1   helpful         4242 non-null   float64
 2   Not helpful     4242 non-null   float64
 3   review_concat   4242 non-null   object 
 4   summary_concat  4242 non-null   object 
 5   overall rating  4242 non-null   int64  
 6   name            4242 non-null   object 
 7   category        4242 non-null   object 
 8   description     4242 non-null   object 
 9   price           4242 non-null   float64
 10  rating          4242 non-null   float64
 11  No_of_Rating    4242 non-null   int64  
dtypes: float64(4), int64(2), object(6)
memory usage: 397.8+ KB


The unnamed: 0 columns can be dropped from both dataset

In [36]:
Amazon_beauty_data= Amazon_beauty_data.drop('Unnamed: 0',1)

In [41]:
Amazon_beauty_data.head()

asin  helpful  Not helpful  \
0  B00KWE08Q0      0.9          0.9   
1  B004W75SLI      1.6          3.2   
2  B00JL2TURM      1.3          1.6   
3  B003WNXZZO      3.8          5.6   
4  B001B7GJ2E      1.7          2.2   

                                       review_concat  \
0  love brush I never tried mini kabuki brush one...   
1  great mask clear skin never tried type mask re...   
2  received product consideration tomoson like id...   
3  unfortunately burn skinnormal rose water doesn...   
4  amazing tool hat like little curve end side ma...   

                                      summary_concat  overall rating  \
0  love brush great brush perfect countouring gre...               5   
1  five star awesome improvement not clear enough...               5   
2  nice natural exfoliant freaking cool amazing w...               5   
3  unfortunately burn skin normal rose water not ...               1   
4  amazing attention mom help teen prevent pockma...               5   

                                                name             category  \
0  Mini Kabuki Makeup Brush Set – Beauty Junkees ...  Tools & Accessories   
1  Origins Clear Improvement Active Charcoal Mask...            Skin Care   
2  pureSOL Konjac Sponge - Activated Charcoal - F...        Personal Care   
3  Shea Terra Organics Beauty Water, Toner & Face...            Skin Care   
4       Body Toolz Double Loop Facial Skin Care Tool  Tools & Accessories   

                                         description  price  rating  \
0  Brand\n\n\nBeauty Junkees\n\n\n\n\nColor\n\n\n...  19.97     4.6   
1  Brand\n\n\nOrigins\n\n\n\n\nSkin Type\n\n\nAll...  14.25     4.5   
2  Color\n\n\nBamboo Charcoal\n\n\n\n\nBrand\n\n\...  10.00     4.2   
3  Ingredients\n\n\nPure Moroccan rose water, Est...  15.75     4.3   
4  Brand\n\n\nBody Toolz\n\n\n\n\nMaterial\n\n\nS...   7.75     4.1   

   No_of_Rating  
0          1366  
1          1082  
2          2466  
3           120  
4           673

# Exploratory Analysis 

The dataframe Amazon_beauty_data contains record from 2014 dataset collected from "http://jmcauley.ucsd.edu/data/amazon/index.html" and scraped data from 2021 which were scraped using selenium webdriver and beautiful soup. There are 4242 records the features in Amazon_beauty_data dataframe are:
1. asin: unique asin id for each product
2. overall rating: the average rating of the products in 2014
3. review_concat: The concatenated reviews of each product that will be used for NLP Sentiment analysis in future
4. helpful and unhelpful: Although it's an average of all the helpful and unhelpful tags to reviews of each product,
it gives an idea of how many people actually go through the reviews and are biased by reviews
5. summary_concat: The heading of each review, will also be used for sentiment analysis
6. name: name of the product
7. category: category of the products. For the sentiment analysis, I will use categories: skincare, makeup and haircare, which has the highest number of records
8. description, price: description and of the product
9. Rating: current rating
10. No_of rating: Number of people who rated the product, gives an idea of how many people purchased the product


In [ ]:
#Number of Products in each category